Python/import.c

```diff
@@ -477,6 +481,7 @@ PyImport_Cleanup(void)
     if (PyDict_CheckExact(modules)) {
         pos = 0;
         while (PyDict_Next(modules, &pos, &key, &value)) {
+            assert(!PyErr_Occurred());
```

Comment:
Neither `PyDict_Next()` nor any other calls before can raise an exception.

Reply:
This is the theory. In practice, something called before might leak an exception. It's to catch bugs "just in case". The assertion is cheap. Debugging an "exception leak" can be very painful.

Reply:
There is no code between this line and previous error check that can raise an exception and don't silence it. I have checked this.